src/spikeinterface/core/waveform_tools.py

```diff
@@ -397,6 +398,280 @@ def _waveform_extractor_chunk(segment_index, start_frame, end_frame, worker_ctx)
                     wfs[pos, :, :] = wf[:, sparsity_mask[unit_ind]]
 
 
+def extract_waveforms_to_single_buffer(
+    recording,
+    spikes,
+    unit_ids,
+    nbefore,
+    nafter,
+    mode="memmap",
+    return_scaled=False,
+    file_path=None,
+    dtype=None,
+    sparsity_mask=None,
+    copy=False,
+    job_name=None,
+    **job_kwargs,
+):
+    """
+    Allocate a single buffer (memmap or or shared memory) and then distribute every waveform into it.
+
+    Contrary to extract_waveforms_to_buffers() all waveforms are extracted in the same buffer, so the spike vector is
+    needed to recover waveforms unit by unit. Importantly in case of sparsity, the channels are not aligned across
+    units.
+
+    Important note: for the "shared_memory" mode wf_array_info contains reference to
+    the shared memmory buffer, this variable must be reference as long as arrays as used.
+    And this variable is also returned.
```

Comment:
Why do you need to keep a reference to the variable? I feel that there is an or missing after "this variable must be referenced* as long as the array is used" because / or " short exaplanation."